src/closed.jl

```diff
@@ -0,0 +1,54 @@
+"""
+Defines the ClosedSet type.
+
+A ClosedSet(left, right) is an interval set where both bounds are closed i.e. the range is [left, right]
```

Comment:
I guess there's a slight conflict with https://github.com/scheinerman/ClosedIntervals.jl/blob/master/src/ClosedIntervals.jl, though. Still, it's undeniably the better term, so I guess we should use module names to disambiguate.